Investigate sqrt_nlp correctness failure in full known-optima suite

## Summary

While validating PR #54, the full known-optima correctness suite exposed a deterministic failure in the existing `sqrt_nlp` case in `python/tests/test_correctness.py`. This appears unrelated to PR #54, which only changes `python/tests/test_pr_correctness.py`, but it should be tracked separately.

## Commands run

```bash
/home/bernalde/.pixi/bin/pixi exec --no-progress -s ipopt -s pkg-config -- \
  env -u CONDA_PREFIX VIRTUAL_ENV=/home/bernalde/repos/discopt/.venv \
  PATH="/home/bernalde/repos/discopt/.venv/bin:$PATH" \
  make test-correctness PYTHON=.venv/bin/python \
  PYTEST=".venv/bin/python -m pytest" \
  MATURIN=".venv/bin/python -m maturin" \
  RUFF=".venv/bin/python -m ruff"
```

Result before interrupting a later long-running JAX compile path:

```text
1 failed, 61 passed in 1400.54s (0:23:20)
FAILED python/tests/test_correctness.py::TestObjectiveCorrectness::test_optimal_value[sqrt_nlp]
```

Direct rerun of the failing case:

```bash
/home/bernalde/.pixi/bin/pixi exec --no-progress -s ipopt -s pkg-config -- \
  .venv/bin/python -m pytest \
  'python/tests/test_correctness.py::TestObjectiveCorrectness::test_optimal_value[sqrt_nlp]' \
  -q --tb=short --timeout=120 --durations=10
```

Result:

```text
FAILED python/tests/test_correctness.py::TestObjectiveCorrectness::test_optimal_value[sqrt_nlp]
AssertionError: [sqrt_nlp] Objective 2.73202901 differs from expected 2.82842712 by 9.64e-02 (tolerance 2.93e-03)
1 failed in 83.43s
```

## Notes

- `python/tests/test_correctness.py` was not modified by PR #54.
- The targeted PR correctness suite and documented PR-fast suite pass with PR #54's changes.
- This needs separate triage to determine whether the `sqrt_nlp` expected optimum, formulation, or solver behavior is wrong.

## Acceptance criteria

- Determine whether `sqrt_nlp`'s analytical expected objective is correct for the modeled problem.
- Determine why the solver returns objective `2.73202901` for this case.
- Update the test, formulation, or solver behavior as appropriate.
- Confirm `make test-correctness` can complete cleanly, or document any remaining long-running blocker separately.

--- a/python/tests/test_correctness.py
+++ b/python/tests/test_correctness.py
@@ -232,10 +232,12 @@ def _build_log_nlp() -> Model:
 def _build_sqrt_nlp() -> Model:
     """min sqrt(x+1) + sqrt(y+1) s.t. x+y>=2, x,y in [0,5].
 
-    sqrt is concave and increasing. On x+y=2:
+    sqrt is concave and increasing, so the minimum occurs on the active
+    constraint at a boundary point. On x+y=2:
     f(x) = sqrt(x+1) + sqrt(3-x). f'(x) = 1/(2*sqrt(x+1)) - 1/(2*sqrt(3-x)) = 0
     => sqrt(3-x) = sqrt(x+1) => 3-x = x+1 => x=1, y=1.
-    obj = sqrt(2) + sqrt(2) = 2*sqrt(2) ~= 2.8284.
+    Since f is concave, that stationary point is the maximum. The minima are
+    at (x, y) = (0, 2) and (2, 0), with obj = 1 + sqrt(3) ~= 2.7321.
     """
     m = dm.Model("sqrt_nlp")
     x = m.continuous("x", lb=0, ub=5)
@@ -562,7 +564,7 @@ INSTANCES: list[ProblemInstance] = [
     ProblemInstance(
         name="sqrt_nlp",
         build_fn=_build_sqrt_nlp,
-        expected_obj=2.0 * math.sqrt(2.0),
+        expected_obj=1.0 + math.sqrt(3.0),
         integer_vars=[],
         bounds={"x": (0, 5), "y": (0, 5)},
         description="min sqrt(x+1)+sqrt(y+1) s.t. x+y>=2",

